### DISCUSSION TITLE
iCub "falling" in Gazebo

### DISCUSSION THREAD

**Discussion by jimmybaraglia** [robotology/community#115] [Q&A] [answered]
Hi,

I am migrating from `iCub_SIM` to **`Gazebo`**, but the iCub appears to "fall" whenever I send a command to the Cartesian controller in `Gazebo`.

I am using the command **`goToPose(xd, od)`** to set the right end-effector to:

```
xd = -0.4 : 0.1 : 0.2
od = -0 : 0.707107 : -0.707107
```

It works very well with the `iCub_SIM`, but not with `Gazebo` (see enclosed screen-shots)

I tried to start the `wholeBodyDynamics` or/and the `gravityCompensator` with the "robot" variable set to  **`icubGazeboSim`**, but it still does the same thing (i.e. falling down).

Any idea why this is happening?

Best,
Jimmy Baraglia

![icub_gazebo](https://cloud.githubusercontent.com/assets/2962132/15283332/a152c8a2-1b84-11e6-9d2d-1dd12afc8613.jpg)
![icub_sim](https://cloud.githubusercontent.com/assets/2962132/15283331/a1529616-1b84-11e6-89e5-886cc602c069.jpg)

**Comment by traversaro** [comment]
I think that the problem is that in Gazebo simulation the velocity control is not properly tested/tuned (see https://github.com/robotology/gazebo-yarp-plugins/issues/240).

This is happening because `simCartesianControl` is using velocity control by default; you can try to make it use the `POSITION_DIRECT` mode by setting the [`PositionControl`](https://github.com/robotology/icub-main/blob/master/app/simCartesianControl/conf/simCartesianLeftArm.ini#L7) option of the `simCartesianControl` to **on**.

**Comment by jimmybaraglia** [comment]
I changed the option PositionControl to "on" in simCartesianRightArm.ini and simCartesianLeftArm.ini.
It worked perfectly, thank you.

Cheers,
Jimmy

**Comment by tkelestemur** [comment]
I still have the falling problem even though I change `PositionControl` to on in both `simCartesianRightArm.ini` and `simCartesianLeftArm.ini`. I am running `tutorial_cartesian_interface` [from icub-tutorials](https://github.com/robotology/icub-tutorials/blob/master/src/motorControlAdvanced/tutorial_cartesian_interface.cpp). After two second running the code iCub falls backwards. 

![screenshot from 2016-06-24 15-31-12](https://cloud.githubusercontent.com/assets/5513783/16337913/9a4e87f6-3a21-11e6-9fa1-9ecec48c646d.png)

**Comment by traversaro** [comment, marked as answer]
By default no balancing controller is running on icub when simulated on Gazebo. To simulate the robot 
as if it was fixed on a fixed base, can you try to launch  the `iCub (fixed)` model? 
If that model is without hands we need to updated it, it is not a big problem.

**Comment by tkelestemur** [comment]
Thank you very much @traversaro, with `iCub (fixed)` it is working now. I wonder why same code causes falling in Gazebo but not in default simulator? From your point is some kind of balancing controller running when we run `iCub_SIM`  ?

**Comment by pattacini** [comment]
@tkelestemur there's no balancing in `iCub_SIM`, and be also careful because `fixed_hip` is `on` by default in `iCub_SIM`; that's why usually you don't see any fall of the iCub therein.

That being said, it would be normal any how to get different results with different simulators.
